include/kernels/staggered_quark_smearing.cuh

```diff
@@ -43,18 +49,23 @@ namespace quda
     int threadDimMapUpper_t0[4];
     int threadDimMapLower_t0[4];
 
-    StaggeredQSmearArg(ColorSpinorField &out, const ColorSpinorField &in, const GaugeField &U, int t0,
-                       bool is_t0_kernel, int parity, int dir, bool dagger, const int *comm_override) :
-      DslashArg<Float, nDim>(out, in, U, in, parity, dagger, false, 3, false, comm_override),
-      out(out, 3),
-      in(in, 3),
-      in_pack(in, 3),
+    StaggeredQSmearArg(cvector_ref<ColorSpinorField> &out, cvector_ref<const ColorSpinorField> &in,
+                       const ColorSpinorField &halo, const GaugeField &U, int t0, bool is_t0_kernel, int parity,
+                       int dir, bool dagger, const int *comm_override) :
+      DslashArg<Float, nDim>(out, in, halo, U, in, parity, dagger, false, improved ? 3 : 1, false, comm_override),
```

Comment:
I think the 3 in all places is intentional---staggered smearing requires a shift-by-2 independent of it being naive or improved staggered, so a halo depth of 1 won't cut it.

What this might mean is the staggered quark smearing code may not work as-is for naive staggered, I'm not sure if that's been properly tested? There may be more hard-coded 3s lurking in some exchange routines in the interface that would make it work.

Reply:
Yeah, agree with your comments here, good catch.  I added a bit too much copy paste here with the implied non-improved, halo-1 additions.  Will remove those.

Reply:
I don't think you meant to push it to `build/include/kernels/staggered_quark_smearing.cuh` in https://github.com/lattice/quda/pull/1465/commits/57913f3db473d35ac96c47b6a8dcac23180f1035 :)

Reply:
🤣

Reply:
7f4694c579199a2c247f1269283b7fcc4ae5f68b